### ISSUE TITLE
silently fails on open()

### ISSUE THREAD

**Issue by nicktime** [pyserial/pyserial#75]
Hi,

when opening file in open() fails, it raises exception - it's fine:
https://github.com/pyserial/pyserial/blob/master/serial/serialposix.py#L295
but when reconfiguring fails, it just silently closes file down - that's puzzling:
https://github.com/pyserial/pyserial/blob/master/serial/serialposix.py#L302

**Comment by zsquareplusc** [comment]
did you experience a problem? or did you just happen to review the code :-)

the `try..except` there is meant to catch any errors in `close` but raise the original exception (see `raise` a few lines below). `close` is called in `open` when reconfigure fails, because users do not expect an open / halfway configured port when the `open` call raises an exception (even when that comes from reconfigure port which open calls).

anyway, in short words: if anything in `open` fails, the port needs to end up closed and and original exception is raised.

**Comment by nicktime** [comment]
Damn it! Didn't notice that raise. Yes, you're right, no problem there. Sorry.
